### PR TITLE
[JENKINS-64119] Display URL Plugin does not use user prefered implementations

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java [hunk 1]
```diff
@@ -80,9 +80,12 @@ DisplayURLProvider lookupProvider() {
     }
 
     @VisibleForTesting
+    @Deprecated
+    /**
+     * @deprecated use
```
Comment (car-roll) [inline src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java:85]:
```suggestion
     * @deprecated use {@link DisplayURLProvider#getUserPReferredProviderProperty}
```

Comment (olamy) [reply src/main/java/org/jenkinsci/plugins/displayurlapi/actions/AbstractDisplayAction.java:85]: oops I forgot that thanks!

**Review by car-roll — APPROVED** [review]

**Review by car-roll — APPROVED** [review]

File: src/test/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProviderTest.java [hunk 2]
```diff
@@ -48,6 +56,163 @@ public void urls() throws Exception {
...
+    }
+
+    @Test
+    public void urlsWithUserDefinedProvider() throws Exception {
+        User foo = User.getById("foo", true);
+        PreferredProviderUserProperty userProperty =
+            new PreferredProviderUserProperty(TestUserDisplayURLProvider.class.getName());
+        foo.addProperty(userProperty);
+        Authentication previous = SecurityContextHolder.getContext().getAuthentication();
+        SecurityContextHolder.getContext().setAuthentication(new UsernamePasswordAuthenticationToken(foo, null));
```
Comment (olamy) [inline src/test/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProviderTest.java:184]: there is maybe something easier to change current user but I was not sure....

File: pom.xml [hunk 3]
```diff
@@ -42,7 +42,6 @@
     <connection>scm:git:git://github.com/jenkinsci/${project.artifactId}-plugin.git</connection>
     <developerConnection>scm:git:git@github.com:jenkinsci/${project.artifactId}-plugin.git</developerConnection>
     <url>http://github.com/jenkinsci/${project.artifactId}-plugin</url>
-    <tag>display-url-api-2.3.3</tag>
```
Comment (dwnusbaum) [inline pom.xml:45]: I think this should be `<tag>HEAD</tag>` for non-release commits, right?

File: src/test/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProviderTest.java (hunk 2 above)
Comment (dwnusbaum) [reply src/test/java/org/jenkinsci/plugins/displayurlapi/DisplayURLProviderTest.java:184]: I think `try (ACLContext c = ACL.as(foo)) { ... code to run with foo as the current user ... }` is the recommended approach.

**Review by dwnusbaum — APPROVED** [review]
Seems fine, but I am not really familiar with this plugin. Based on the description of [JENKINS-64119](https://issues.jenkins.io/browse/JENKINS-64119) I was expecting a one-line patch to change [`EnvironmentContributorImpl`](https://github.com/jenkinsci/display-url-api-plugin/blob/8a5092a0c6f7a468c85a95a69314d6855f45c80a/src/main/java/org/jenkinsci/plugins/displayurlapi/EnvironmentContributorImpl.java#L32) to use `getDefault` instead of `get`, but I guess the issue that neither of those methods respected the per-user preference before this PR?

File: pom.xml (hunk 3 above)
Comment (olamy) [reply pom.xml:45]: HEAD is default value :)

Comment (dwnusbaum) [reply pom.xml:45]: Today I learned 😄

**Comment by olamy** [comment]
one line patch is breaking backward compat :( finally there is more than one way to get this..... especially looking at how is done in `AbstractDisplayAction`

**Comment by oleg-nenashev** [comment]
I will try to test the patch next week. Thanks @olamy !

**Review by oleg-nenashev — APPROVED** [review]
I have tested it locally with Jenkinsfile Runner, and I confirm that the patch works well.
Thanks @olamy et all!

**Pull request merged**